### PR TITLE
[Prepare for go1.20] *: Bump versions and fix tests

### PR DESCRIPTION
**Pull request by MadhavJivrajani**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

Commit 1:

Bumping version of govmomi to include changes that
better handle TLS errors. Bump nescessary
to prepare for when the version of Go is
bumped to 1.20.

Command run:
```
./hack/pin-dependency.sh github.com/vmware/govmomi v0.30.0 
```

See https://github.com/kubernetes/kubernetes/pull/114502#issuecomment-1369026247

Commit 2:

This PR also adapts to this bump.

Commit 3:

This commit modifies the unwrap logic of tests for vclib connection tests to accommodate for new error types introduced in go1.20 and be backward compatible with go1.19

#### Special notes for your reviewer:

~On running the canary job for unit tests, the `TestWithVerificationWithoutCaCertOrThumbprint` test will still fail. See https://github.com/kubernetes/kubernetes/pull/114502#issuecomment-1369030189~

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

/priority important-soon
/assign @liggitt @dims 

(I messed up something in git because of which the previous PR closed, apologies!)

### CODE REVIEW

**Comment by MadhavJivrajani** [comment]
/test pull-kubernetes-unit-go-canary

The `TestWithVerificationWithoutCaCertOrThumbprint` test will still fail.
See https://github.com/kubernetes/kubernetes/pull/114502#issuecomment-1369030189

**Comment by liggitt** [comment]
xref https://github.com/kubernetes/release/issues/2815

**Comment by liggitt** [comment]
hmm... this thinks total dependencies increased by 4:

```diff
 {
-	"directDependencies": 212,
-	"transitiveDependencies": 234,
-	"totalDependencies": 285,
+	"directDependencies": 213,
+	"transitiveDependencies": 238,
+	"totalDependencies": 289,
 	"maxDepthOfDependencies": 23
 }
```

**Comment by liggitt** [comment]
hmm... govmomi already includes 6700+ init() functions (80+% of the init functions in kubernetes), and this library bump adds ~470+ more (search "+func init()" in https://patch-diff.githubusercontent.com/raw/kubernetes/kubernetes/pull/114766.diff)

the unfortunate thing is that those run on every startup of every binary they are linked into, even ones which don't make use of vmware functionality at all :-/

**Comment by liggitt** [comment]
new deps this adds to the dependency tree:
* github.com/a8m/tree
* github.com/dougm/pretty
* github.com/pkg/diff
* github.com/vmware/vmw-guestinfo

deps this drops from the dependency tree:
* gopkg.in/errgo.v2

**Comment by liggitt** [comment]
After chasing https://github.com/kubernetes/kubernetes/pull/114766#issuecomment-1370164010, I think that's spurious... need to dig more, but I don't think that blocks this merging

I'm not thrilled with https://github.com/kubernetes/kubernetes/pull/114766#issuecomment-1370168605, especially since we'll likely need to backport this to release branches in order to move them to go1.20. That said, I don't see a quick fix to that.

**Comment by MadhavJivrajani** [comment]
@liggitt done.
/test pull-kubernetes-unit-go-canary

**Comment by liggitt** [comment]
both unit tests flaked on https://github.com/kubernetes/kubernetes/pull/114678#issuecomment-1374075860

/retest

**Comment by liggitt** [comment]
/test pull-kubernetes-dependencies
to run check added in https://github.com/kubernetes/kubernetes/pull/114952

File: staging/src/k8s.io/legacy-cloud-providers/vsphere/vsphere_test.go [hunk 1]
```diff
@@ -165,7 +165,8 @@ func configFromSimWithTLS(tlsConfig *tls.Config, insecureAllowed bool) (VSphereC
 	model.Service.ServeMux.Handle(path, handler)
 
 	// vAPI simulator
-	path, handler = vapi.New(s.URL, vpx.Setting)
+	paths, handler := vapi.New(s.URL, vpx.Setting)
+	path = paths[0]
```
[the remark this answers is not in the document]
Comment (MadhavJivrajani) [reply staging/src/k8s.io/legacy-cloud-providers/vsphere/vsphere_test.go:169]: I had discarded them, iirc according to my understanding `paths[0]` is what was initially returned from `vapi.New` before the API was changed:

https://github.com/vmware/govmomi/blob/2bd857b669cb777eb13aaa90e9d49b3f2e39081a/vapi/simulator/simulator.go#L182

The commit that made this change: https://github.com/vmware/govmomi/commit/2017e846082636c8910a54f9ffaec5c31475e8b4

**Pull request merged**